pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -983,6 +985,26 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, allowEmul
 		}
 	}
 
+	ifaceNamingScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
+	for _, network := range netvmispec.NetworksToHotplugWhosePodIfacesAreReady(vmi) {
+		log.Log.Infof("will hot plug %s", network.Name)
+		podIfaceName, wasFound := ifaceNamingScheme[network.Name]
+		if !wasFound {
+			return nil, fmt.Errorf("could not find the pod interface name for network [%s]", network.Name)
+		}
+
+		ifaceXML, err := xml.Marshal(converter.DomainInterfaceSpec(c, network.Name, virtnetlink.GenerateTapDeviceName(podIfaceName)))
+		if err != nil {
+			log.Log.Warningf("failed to marshall the domain interface spec for hotplugging the %s attachment", network.Name)
+			continue
+		}
+
+		if err := dom.AttachDevice(strings.ToLower(string(ifaceXML))); err != nil {
```

Comment:
If I am not mistaken, at this stage, the interface may be already attached and this will wrongly error.

Also, the operation is pretty costly and may block the virt-handler for some time.
We had the same need when treating the SR-IOV interface and [this was the solution](https://github.com/kubevirt/kubevirt/blob/5588ee3a969d0ecd433b79d59bcd61ccb171e05a/pkg/virt-launcher/virtwrap/manager.go#L344). Can you just use the same logic here?

Reply:
> Do you have any figures to back this up ?

The operation of attaching to the domain is intrusive and has IO cost.
It requires inter process communication and libvirt processing to execute the operation on the domain which blocks until completed.
I have done no measurements of the cost, nor I think we need to when the peer is a process we do not control its behavior.

> I can, but I'm not convinced this issue can occur in the first place. I'd rather not add code that is not needed.

The aync behavior of the reconcile and feedback of the virt-launche to the virt-handler is given.
I have tried to explain why it is incorrect to check the current status in the virt-launcher using data calculated by the virt-handler.
I can try again, please point me to where you think it is wrong:
- t1: Attachment is triggered.
- t2: Event arrives to the virt-handler for the same VMI, either  due to virt-launcher update or because something changed on the VMI.
- t3: Attachment (that started at t1) finishes.
- t4: virt-handler processes the event from t2, now being aware yet anything changed on the domain.
- t5: SyncVMI is called and due to the same data, attachment is triggered again, failing the SyncVMI.
- t6: Event from libvirt reaches virt-handler, which sends it to virt-handler.
....

If the check would have taken into concideration the actual current state of the domain (which you do have), no failure would have occured.
Yes, you could depend on the reconcile to fix it eventually, but IMO this nasty. You have all that is needed to avoid it.
Frankly, it also makes more sense to me that the check will be against the current state that is computed from the domain, not on a state computed by a different component.
I now realize this is one advantage of a dedicate command over re-using the SyncVMI, to have a cleaner separation. But I think we do not need to involve another topic into this, the existing logic can be improved with this SyncVMI as is.

If you do not want to fix it now, please keep it open so I can take care of it in a follow up.

Reply:
OK then. IMO, it is preferable to move forward with this and improve later.